docs/aws/audit/snsmonitoring/rules/is_sns_topic_encrypted_remediation.mdx

```diff
@@ -1,6 +1,195 @@
+               sns_client.set_topic_attributes(
+                   TopicArn=topic_arn,
+                   AttributeName='KmsMasterKeyId',
+                   AttributeValue='alias/aws/sns'  # Use the default AWS managed key for SNS
+               )
+           else:
+               print(f"SNS topic {topic_arn} is already encrypted.")
+
+   if __name__ == "__main__":
+       enforce_encryption_on_sns_topics()
+   ```
+
+4. **Run the Script**:
+   Execute the script to ensure all your SNS topics are encrypted.
+   ```bash
+   python enforce_sns_encryption.py
+   ```
+
+This script will list all SNS topics in your AWS account, check if they are encrypted, and if not, it will apply encryption using the default AWS managed key for SNS (`alias/aws/sns`). You can modify the `AttributeValue` to use a custom KMS key if needed.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Simple Notification Service (SNS) dashboard by selecting "Services" from the top menu, then typing "SNS" in the search box and selecting "Simple Notification Service".
+3. In the SNS dashboard, select "Topics" from the left-hand navigation pane. This will display a list of all the SNS topics in your AWS account.
+4. For each topic, check the "Encryption" column. If the column is blank or shows "Disabled", then the SNS topic is not encrypted. If it shows "Enabled", then the topic is encrypted.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the SNS topics in your AWS account. You can do this by using the following AWS CLI command:
+
+```bash
+aws sns list-topics
+```
+
+This command will return a list of all SNS topics in your AWS account.
+
+2. Once you have the list of SNS topics, you can check the encryption status of each topic. You can do this by using the following AWS CLI command:
+
+```bash
+aws sns get-topic-attributes --topic-arn <TopicARN>
+```
+
+Replace `<TopicARN>` with the ARN of the SNS topic you want to check. This command will return the attributes of the specified SNS topic.
+
+3. In the output of the above command, look for the `KmsMasterKeyId` attribute. If this attribute is present and has a value, it means that the SNS topic is encrypted.
+
+4. Repeat steps 2 and 3 for each SNS topic in your AWS account. If any SNS topic does not have the `KmsMasterKeyId` attribute or if this attribute does not have a value, it means that the SNS topic is not encrypted.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so. You can install it using pip:
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>